metpy/calc/thermo.py

```diff
@@ -875,7 +875,7 @@ def specific_humidity_from_mixing_ratio(mixing_ratio):
     mixing_ratio, mixing_ratio_from_specific_humidity
 
     """
-    return mixing_ratio / (1 + mixing_ratio)
+    return mixing_ratio.to('dimensionless') / (1 + mixing_ratio.to('dimensionless'))
```

Comment:
I think this is overkill, since you'll be dividing `mixing_ratio` by itself and cancelling the units. It's also a change to behavior, since previously you could pass plain numbers to this function (provided they were properly dimensionless) and get correct behavior; this now assumes units.

If this is a problem you've encountered, I'm open to a test case showing so.

Reply:
You are right that units cancel, but mixing ratio has to be in kg/kg (or g/g) here. If you pass g/kg, the result will be incorrect. So I think it's better to enforce correct units in these formulas.

Reply:
I'd like to keep the ability to pass in values without units. So instead of doing the conversion inline, can you instead add a block at the start of the function:
```python
try:
    mixing_ratio = mixing_ratio.to('dimensionless')
except AttributeError:
    pass
```